forte/data/ontology/top.py

```diff
@@ -686,5 +688,169 @@ def get(
+        within the grid cell will be preserved.
+
+
+        Args:
+            h_idx: the index of the grid cell of the first dimension.
+            w_idx: the index of the grid cell of the second dimension.
+            image_payload_idx: An integer that represents the index of
+                the image in the payloads. If it's not None, it will use the
+                image from the corresponding payload. Otherwise,
+                it will use the image
+                from the payload of ``self._image_payload_idx``.
+
+        Raises:
+            ValueError: ``h_idx`` is out of the range specified by ``height_n_width``.
+            ValueError: ``w_idx`` is out of the range specified by ``height_n_width``.
+
+        Returns:
+            numpy array that represents the grid cell.
+        """
+        if not 0 <= h_idx < self.height:
+            raise ValueError(
+                f"input parameter h_idx ({h_idx}) is"
+                "out of scope of h_idx range"
+                f" {(0, self.height)}"
+            )
+        if not 0 <= w_idx < self.width:
+            raise ValueError(
+                f"input parameter w_idx ({w_idx}) is"
+                "out of scope of w_idx range"
+                f" {(0, self.width)}"
+            )
+
+        if image_payload_idx is None:
+            if self._image_payload_idx is None:
+                raise ValueError(
+                    "There is not image_payload_idx associate "
+                    "with the Grids instance neither"
+                    " image_payload_idx in the ``get_grid_cell`` "
+                    "function parameters. Please make sure either "
+                    "one of them is valid."
+                )
+            image_payload_idx = self._image_payload_idx
+
+        img_arr = self.pack.get_image_array(image_payload_idx)
+        c_h, c_w = (
+            img_arr.shape[0] // self.height,
+            img_arr.shape[1] // self.width,
+        )
+        array = np.zeros(img_arr.shape)
+        array[
+            h_idx * c_h : (h_idx + 1) * c_h, w_idx * c_w : (w_idx + 1) * c_w
+        ] = img_arr[
+            h_idx * c_h : (h_idx + 1) * c_h, w_idx * c_w : (w_idx + 1) * c_w
+        ]
+        return array
+
+    @property
+    def image_payload_idx(self) -> int:
+        if self._image_payload_idx is None:
+            raise ValueError(
+                "This Grids instance is not associated with "
+                "any image payload yet."
+            )
+        return self._image_payload_idx
+
+    @property
+    def num_grid_cells(self):
+        return self.height * self.width
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return self.image_payload_idx == other.image_payload_idx
+
+    def __hash__(self) -> int:
+        r"""
+        The hash function for ``Sketch`` class with a numpy array as a class
+        attribute.
+        """
+        return hash((self._image_payload_idx, self._tid))
```

Comment:
I feel `tid` is enough

Reply:
Maybe I should remove this method and just use [`hash()`](https://github.com/asyml/forte/blob/8af7f5acd8e89efe7393b2e5a633e0c076e3e7b4/forte/data/ontology/core.py#L398) from it's parent class.